src/autorun.ts

```diff
@@ -0,0 +1,174 @@
+	const installContractsToRun = installContracts.filter(isInstallContractNotExecuted);
+	const contractsPromises = installContractsToRun.map(runInstallContractData);
+	await Promise.allSettled(contractsPromises);
+}
+
+/* Definitions */
+
+async function getContractPathsInDirectory(directoryPath: string): Promise<string[]> {
+	const directory = await fs.readdir(directoryPath);
+	return Array.from(directory.entries())
+		.map((entry) => entry[1])
+		.filter((name) => isZencodeFile(name))
+		.map((name) => join(directoryPath, name));
+}
+
+async function loadContractsData(directoryPath: string): Promise<ContractData[]> {
+	const contractPaths = await getContractPathsInDirectory(directoryPath);
+	const contractDataPromises = contractPaths.map(loadContractData);
+	const maybeContractData = await Promise.all(contractDataPromises);
+	return maybeContractData.filter((d) => Boolean(d)) as ContractData[];
+}
+
+async function runContractData({ contract, keys, conf, data, path }: ContractData) {
+	try {
+		const s = new Slangroom([http, wallet]);
+		const { result } = await s.execute(contract, { keys, data, conf });
+		L.info('Run contract');
+		L.info(path);
+		L.info(result);
+	} catch (e) {
+		L.error(e);
+	}
+}
+
+async function runInstallContractData(contractData: InstallContractData) {
+	try {
+		await runContractData(contractData);
+		await decommissionInstallContract(contractData);
+	} catch (e) {
+		L.error(e);
+	}
+}
+
+//
+
+export interface ContractData {
+	path: string;
+	contract: string;
+	keys?: Record<string, unknown> | undefined;
+	data?: Record<string, unknown> | undefined;
+	metadata?: Record<string, unknown> | undefined;
+	conf?: string | undefined;
+}
+
+async function loadContractData(contractPath: string): Promise<ContractData | undefined> {
+	try {
+		const contract = await readContract(contractPath);
+		if (!contract) throw new Error('Contract not found');
+
+		const { name: contractName, dir: directoryPath } = parse(contractPath);
+
+		const keysPath = join(directoryPath, `${contractName}.keys.json`);
+		const keys = await readJsonObject(keysPath);
+
+		const dataPath = join(directoryPath, `${contractName}.data.json`);
+		const data = await readJsonObject(dataPath);
+
+		const metadataPath = join(directoryPath, `${contractName}.metadata.json`);
+		const metadata = await readJsonObject(metadataPath);
+
+		const confPath = join(directoryPath, `${contractName}.conf`);
+		const conf = await readFileContent(confPath);
+
+		return {
+			path: contractPath,
+			contract,
+			keys,
+			conf,
+			data,
+			metadata
+		};
+	} catch (e) {
+		L.error(e);
+		return undefined;
+	}
+}
+
+//
+
+async function readContract(path: string): Promise<string | undefined> {
+	const contract = await readFileContent(path);
+	return contract ? formatContract(contract) : undefined;
+}
+
+function formatContract(baseContract: string): string {
+	return `Rule unknown ignore\nRule check version ${config.zenroomVersion}\n${baseContract}`;
```

Comment:
this is also done in another place (src/directory) please keep them in sync by defining a costant or a function

Reply:
Done in: https://github.com/ForkbombEu/ncr/pull/42/commits/abf97395ec620c8762084f0c82ab3a2e6e8d3335